### PR TITLE
feat(insights): keep configuration when navigating between insights

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

Users want to switch from one query to another while keeping relevant configuration.

## Changes

This PR re-adds said functionality.

### (Potential) Follow-ups
- An error is thrown when switching from trends -> funnels -> trends with the compare option enabled. The reason for this is that the results are still based on the funnel response for a brief moment. Once this is fixed, the compare option can be enabled here as well.
- Event and action nodes include math properties which are only supported in some insights e.g. trends, but not in others e.g. funnels. Not all math properties supported in trends are supported in stickiness insights e.g. count per user average. Should we have different types for them and convert this accordingly? We seem to handle this well backend side, but frontend side is broken in some cases e.g. stickiness insight with unsupported math type.
In the same spirit retentionFilter.target_entity and returning_entity should not have an order property.

## How did you test this code?

Added tests and locally, with 👀

### CODE REVIEW

File: frontend/__snapshots__/components-cards-insight-card--insight-card.png
Comment (Twixes) [inline frontend/__snapshots__/components-cards-insight-card--insight-card.png:1]: There's a new "Country Name count" now 🤔 Don't think this is expected

**Review by Twixes — COMMENTED** [review]
I like the approach here. Just found one bug: The configuration is kept as expected when a saved insight is loaded and immediately switched to a different type – but it's not kept when I first load the saved insights page and get to that same insight _from there_.

File: frontend/__snapshots__/components-cards-insight-card--insight-card.png
Comment (thmsobrmlr) [reply frontend/__snapshots__/components-cards-insight-card--insight-card.png:1]: This is unrelated to the changes here, as the "Country Name count" exists in the current version of Storybook as well. See https://storybook.posthog.net/?path=/story/components-cards-insight-card--insight-card. I'm looking into a fix separately.

**Comment by thmsobrmlr** [comment]
Awesome catches @Twixes, thanks for the thorough look!

> Just found one bug: The configuration is kept as expected when a saved insight is loaded and immediately switched to a different type – but it's not kept when I first load the saved insights page and get to that same insight from there.

This is because the `insightDataLogic` isn't mounted yet when `insightLogic` calls `setInsight` and so the respective listener doesn't fire. I've added a query cache update in `insightNavLogic`'s `afterMount`.

There'd be an alternative approach that removes the problematic code altogether, simplifying a bit what happens when loading an insight. I've been trying it out here https://github.com/PostHog/posthog/pull/18346, but am a bit worried about unintended effects of this.

**Review by Twixes — APPROVED** [review]

**Pull request merged**